plugin/pkg/scheduler/scheduler.go

```diff
@@ -89,6 +92,12 @@ func (s *Scheduler) Run() {
 
 func (s *Scheduler) scheduleOne() {
 	pod := s.config.NextPod()
+
+	// If the assumed pods store contains this pod there is a conflict,
+	// since the store can't contain a pod without a host, and NextPod
+	// only gives us pods without hosts.
+	s.config.Modeler.ForgetAssumedPod(pod)
```

Comment:
Oh! btw, it will fail a fit predicate on the host port because MapPodsToMachines uses the modeler as a podlister (https://github.com/GoogleCloudPlatform/kubernetes/blob/master/pkg/scheduler/predicates.go#L354). So it could be the same thing.